src/utils/hooks/useEventHandler.ts

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export function useEventHandler<T extends Function>(handler?: T) {
+    const ref = React.useRef<T | undefined>(handler);
+    React.useLayoutEffect(() => {
+        ref.current = handler;
+    }, [handler]);
+    // @ts-expect-error
+    return React.useCallback<T>((...args) => {
+        return ref.current?.(...args);
+    }, []);
+}
```

Comment:
Can you please explain, how does that works? And why it's better, that callback, that were previously?

Reply:
The hook returns a stable function (an empty list of dependencies), but this function always calls the actual function associated with the last render. The returned function should be used as an event handler or inside a useEffect. This PR uses it to set event handlers in the Monaco editor. Previously setState + useEffect were used to achieve the same effect. It was difficult to read and understand.

Reply:
Thank you, I got it. I think it will be good to leave your comment in the code as well (for `useEventHandler`):
`The hook returns a stable function (an empty list of dependencies), but this function always calls the actual function associated with the last render. The returned function should be used as an event handler or inside a useEffect`